src/pydicom/pixels/decoders/base.py

```diff
@@ -1077,19 +1102,26 @@ def _as_array_encapsulated(runner: DecodeRunner, index: int | None) -> "np.ndarr
         if index is not None:
             # The decoding plugin may alter runner.bits_allocated to give a
             #   different dtype itemsize
-            arr[:] = np.frombuffer(runner.decode(index=index), dtype=runner.pixel_dtype)
+            frame = np.frombuffer(runner.decode(index=index), dtype=runner.pixel_dtype)
             runner.set_option("bits_allocated", original_bits_allocated)
 
+            if runner.get_option("is_bitpacked"):
+                frame = unpack_bits(frame)[:pixels_per_frame]
```

Comment:
`unpack_bits()` takes `bytes`, but `frame` is an `ndarray`.

I'm kind of surprised mypy isn't flagging this.

Reply:
Hmm yes. This does what is supposed to do because inside `unpack_bits` an array is created from `frame` anyway using `np.frombuffer`, and it seems that calling `np.frombuffer(arr, dtype=arr.dtype)` for any array results in a copy of `arr` being returned.

But it would be best not to ignore the type hints and rely on this non-intuitive behaviour. Refactored in [6d97c96](https://github.com/pydicom/pydicom/pull/2216/commits/6d97c96a2bb7d5b77dfe3ee9baacc97bd5f08699) to be more explicit.

Reply:
Thanks for catching this, I had not noticed the discrepancy